src/gui/qgsmaptooladvanceddigitizing.cpp

```diff
@@ -34,6 +34,11 @@ void QgsMapToolAdvancedDigitizing::canvasPressEvent( QgsMapMouseEvent *e )
   {
     mCadDockWidget->applyConstraints( e );  // updates event's map point
 
+    if ( mCadDockWidget->mapPointMatch().hasVertex() )
+    {
+      e->snapPoint();
```

Comment:
If I understand correctly, the idea is to force snapping to a point (and return to original point if failed).
can you explain the reasoning? Mainly which scenario.

Reply:
When you activate the CadDock, it only returns the 2D point, by doing this if the vertex is identified, it can return a 3D point